File: CS/WindowsApplication1/HelpHelper.cs
using System;
using System.Text;
using System.Windows.Forms;
using System.Drawing;
using DevExpress.Utils.Drawing.Helpers;

namespace WindowsFormsApplication3 {
    public interface IHelpOwner {
        void DoBaseWndProc(ref Message msg);
        void DoHelpRequested();
        Control Control { get; }
    }
    public class HelpHelper {
        IHelpOwner owner;
        bool helpRequested = false;
        public HelpHelper(IHelpOwner owner) {
            this.owner = owner;
        }
        public bool DoWndProc(ref Message msg) {
            if (msg.Msg == MSG.WM_NCLBUTTONDOWN) {
                Point pt = new Point(msg.LParam.ToInt32());
                //Console.WriteLine(string.Format("{0}, {1}", msg.WParam, pt.ToString()));
                if (msg.WParam.ToInt32() == DevExpress.Utils.Drawing.Helpers.NativeMethods.HT.HTHELP) {
                    this.helpRequested = false;
                    owner.DoBaseWndProc(ref msg);
                    return true;
                }
            }
            if (msg.Msg == 0x210/*wm_parentnotify*/ && msg.WParam.ToInt32() == MSG.WM_LBUTTONDOWN) {
                if (!helpRequested) {
                    owner.Control.BeginInvoke(new MethodInvoker(CheckShowHelp));
                }
            }
            return false;
        }
        void CheckShowHelp() {
            if (!helpRequested) {
                this.helpRequested = true;
                owner.DoHelpRequested();
            }
        }
        public void SetHelpRequested() { this.helpRequested = true; }
    }
}

File: VB/WindowsApplication1/HelpHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Text
Imports System.Windows.Forms
Imports System.Drawing
Imports DevExpress.Utils.Drawing.Helpers

Namespace WindowsFormsApplication3
	Public Interface IHelpOwner
		Sub DoBaseWndProc(ByRef msg As Message)
		Sub DoHelpRequested()
		ReadOnly Property Control() As Control
	End Interface
	Public Class HelpHelper
		Private owner As IHelpOwner
		Private helpRequested As Boolean = False
		Public Sub New(ByVal owner As IHelpOwner)
            Me.owner = owner
		End Sub
		Public Function DoWndProc(ByRef message As Message) As Boolean
            If message.Msg = MSG.WM_NCLBUTTONDOWN Then
                Dim pt As New Point(message.LParam.ToInt32())
                'Console.WriteLine(string.Format("{0}, {1}", msg.WParam, pt.ToString()));
                If message.WParam.ToInt32() = DevExpress.Utils.Drawing.Helpers.NativeMethods.HT.HTHELP Then
                    Me.helpRequested = False
                    owner.DoBaseWndProc(message)
                    Return True
                End If
            End If
            If message.Msg = &H210 AndAlso message.WParam.ToInt32() = MSG.WM_LBUTTONDOWN Then
                If message.Msg = &H210 AndAlso message.WParam.ToInt32() = MSG.WM_LBUTTONDOWN Then
                    If (Not helpRequested) Then
                        owner.Control.BeginInvoke(New MethodInvoker(AddressOf CheckShowHelp))
                    End If
                End If
            End If
            Return False
        End Function
		Private Sub CheckShowHelp()
			If (Not helpRequested) Then
				Me.helpRequested = True
				owner.DoHelpRequested()
			End If
		End Sub
		Public Sub SetHelpRequested()
			Me.helpRequested = True
		End Sub
	End Class
End Namespace
